pkg/apis/agones/v1/gameserver.go

```diff
@@ -437,6 +442,30 @@ func (gs *GameServer) applyEvictionStatus() {
 	}
 }
 
+func (gs *GameServer) applyCountsListsStatus() {
+	if !runtime.FeatureEnabled(runtime.FeatureCountsAndLists) {
+		return
+	}
+	// Set status to prevent panicking
+	if gs.Status.Counters == nil && gs.Spec.Counters != nil {
+		gs.Status.Counters = make(map[string]CounterStatus)
+	}
+	if gs.Status.Lists == nil && gs.Spec.Lists != nil {
+		gs.Status.Lists = make(map[string]ListStatus)
+	}
```

Comment:
Is this logic needed? Does `gs.Status.Counters = gs.Spec.Counters` simply work? I'm not sure how the deepcopy works but I'm taking `gs.applyEvictionStatus()` as reference.